File: test_samples/sample_vuln.py
# Sample vulnerable file for testing TaintFlow
import sqlite3
from flask import request

def get_user_data():
    # Source: request.args (CWE-89)
    username = request.args.get('username')
    conn = sqlite3.connect("users.db")
    cursor = conn.cursor()
    # SQL injection vulnerability (CWE-89)
    query = "SELECT * FROM users WHERE username = '" + username + "'"
    cursor.execute(query)
    return cursor.fetchall()

def authenticate():
    # Hardcoded credential (CWE-798)
    api_token = "super_secret_token_12345678"
    print("Token initialized")

File: test_samples/SampleVuln.java
import javax.servlet.http.HttpServletRequest;
import java.sql.Connection;
import java.sql.Statement;

public class SampleVuln {
    public void doGet(HttpServletRequest request, Connection conn) throws Exception {
        // Source: request.getParameter (CWE-89)
        String username = request.getParameter("username");
        
        // SQL injection vulnerability (CWE-89)
        String query = "SELECT * FROM users WHERE username = '" + username + "'";
        
        Statement stmt = conn.createStatement();
        stmt.execute(query);
    }

    public void authenticate() {
        // Hardcoded credential (CWE-798)
        String adminToken = "super_secret_java_token_987654321";
        System.out.println("Authenticated: " + adminToken);
    }
}
